src/error/MatterError.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { StatusCode } from "../matter/interaction/InteractionMessages";
+
+/** Error base class for all errors thrown by this library. */
+export class MatterError extends Error {}
+
+/** Specific Error for when a fabric is not found. */
+export class FabricNotFoundError extends MatterError {}
+
+
+/** Error base Class for all errors related to the status response messages. */
+export class StatusResponseError extends MatterError {
+    private static errorMap = new Map<StatusCode, new(...args: any[]) => StatusResponseError>();
```

Comment:
This is an non-standard way of doing a mapping.

I am not sure where exactly you need the mapping (expect in the test where you can directly grab what you want), but if you do, a more standard way woudl be:
const StatusError: { [statusCode: StatusCode]: StatusResponseError } = {
     [StatusCode.Failure]: FailureStatusResponseError;
     (etc...)
};

Or you can simply add the status code inside the StatusResponseError and not worry about individual naming.

Reply:
Needed to be able to throw  "the exception that matches to the statusresponse coming in ... will move ro yur approach

Reply:
But ok you are right we can also make it very generic then we just thor an StatusResponse exception with the statuscode ... so catching needs to check instance and then the code ... also works. will adjust